tests/test_batch_executor.py

```diff
@@ -0,0 +1,121 @@
+import random
```

Comment:
We didn't have Oven tests so I figured I would kick things off with a couple tests, especially concerning with what I added in this PR

Reply:
I don't even know what an oven is 🤷